### DISCUSSION TITLE
Is there a way to access qutebrowser history in userscipt?

### DISCUSSION THREAD

**Discussion by hehelego** [qutebrowser/qutebrowser#7142] [Q&A] [answered]
Hey guys, I want to implement a userscript which leverage the power of `fzf` to quickly navigate through the browsing history.  
However, I found that we can not directly access the history in userscript environment.  
So is there a convenient way to do that?

Also, I do noticed that the history is stored in `$QUTE_DATA_DIR/history.sqlite`,
I am wondering whether or not the table structure is fixed.

**Comment by mschilli87** [comment, marked as answer]
AFAIK, reading the data from `$QUTE_DATA_DIR/history.sqlite` is your best option. I don't think there is a guarantee for the format to remain stable but IIRC it hasn't changed in a long time. So hopefully, by the time your script breaks, there will already be the plugin API to replace that hack.

**Reply by hehelego** [reply]
thanks.
